### PR TITLE
Add const expression utility to binaryen-c/.js

### PR DESCRIPTION
**Pull request by dcodeIO**
Given an i32/i64/f32/f64.const expression reference, these functions return the expression's literal value. One use case is precomputing an expression (i.e. through running the precompute pass) and then checking the precomputed value.

### CODE REVIEW

File: src/binaryen-c.h [hunk 1]
```diff
@@ -371,6 +371,18 @@ BinaryenExpressionId BinaryenExpressionGetId(BinaryenExpressionRef expr);
 BinaryenType BinaryenExpressionGetType(BinaryenExpressionRef expr);
 // Print an expression to stdout. Useful for debugging.
 void BinaryenExpressionPrint(BinaryenExpressionRef expr);
+// Gets the 32-bit integer value of the specified known-to-be-constant expression.
+int32_t BinaryenConstGetValueI32(BinaryenExpressionRef expr);
+// Gets the 64-bit integer value of the specified known-to-be-constant expression.
+int64_t BinaryenConstGetValueI64(BinaryenExpressionRef expr);
+// Gets the low 32-bits of a 64-bit integer value of the specified known-to-be-constant expression.
```
Comment (kripken) [inline src/binaryen-c.h]: i guess that low/high variants are for calling from JS? worth mentioning that i think.

File: src/binaryen-c.h [hunk 2]
```diff
@@ -371,6 +371,18 @@ BinaryenExpressionId BinaryenExpressionGetId(BinaryenExpressionRef expr);
 BinaryenType BinaryenExpressionGetType(BinaryenExpressionRef expr);
 // Print an expression to stdout. Useful for debugging.
 void BinaryenExpressionPrint(BinaryenExpressionRef expr);
+// Gets the 32-bit integer value of the specified known-to-be-constant expression.
```
Comment (kripken) [inline src/binaryen-c.h]: "known-to-be-constant" seems to use "constant" as a general property - we should say that the expression passed in must be a captital-C `Const` node.

**Comment by dcodeIO** [comment]
Still thinking how to implement a `BinaryenExpressionPrecompute` function by reusing the precompute pass. That would be super useful.

**Comment by kripken** [comment]
Why not just run the precompute pass on the whole module? Is there a reason to do it on individual expressions?

**Comment by dcodeIO** [comment]
Yeah, the reason is that the MVP does not allow more complex (for example binary) expressions as initializers of, let's say, globals and I am trying to avoid reimplementing precompute in my compiler. Hence, if I have a const global with an initializer like `1 << 2`, I can't use it as is, but must instead make the global mutable and initialize it in the start function.

Being able to precompute just the initializer would solve this, while not affecting the entire module.

**Comment by kripken** [comment]
I see. One way could be, in JS, to create a module (empty except you would want the global imports), add just that expression in a function, and run the pass on that, then get that function body. We should make doing stuff like that easy anyhow from JS, I think, so this could be a good way to verify that.

**Comment by dcodeIO** [comment]
Sure, that would be one option. This will, however, not cover things like `globalB = 1 + globalA`.

**Comment by kripken** [comment]
I don't see why not, can you elaborate? (Or do you mean you'd need to loop, continuing to work while something is reduced? Yes, that would be necessary.)

**Comment by dcodeIO** [comment]
Well, because `globalA` then doesn't exist in the empty module. Got something working now, though, will create a PR soon. Simply reuses `precomputeValue` of `Precompute`.

**Pull request merged**